### PR TITLE
Automate nightly mass download

### LINKED ISSUE #62: Automating the download of mass data
One issue that we are always facing is how to automatically download data from mass onto jasmin. We already have the automatically generated scripts in `mass_scripts`, but they need to be run manually. Automating it would be nice.

In V1, we had a rather frustrating process of:
- Set up a cron job on my PML desktop. 
- The cron script was procedurally generated from a manually curated list of "live" jobs.
- The cron job runs every night until it is replaced or manually edited.
- Each jobID had it own line in the cronscript - which were run once per hour.
- Each job was sent to the login machine, which then lauched a second ssh command to mass-cli1
- Then fainlly on mass-cli1, the moo get script was executed (via a Python script!)

I've asked Jasmin helpdesk for their thoughts on how to stream line this process. They said that we can add one line to the mass-cli1 cron table! Yikes! 

So, here are my thoughts:
- The one line on the cron table should activate an executable that lives in a shared bgcval space 
    -  ie `/gws/nopw/j04/esmeval/bgcval2/shared_mass_cron.sh`
    - I guess a copy of this file can also live in `bgcval2/scripts` or something - so it's at least version controlled. 
- This executable, `shared_mass_cron.sh` finds and runs everyone's `moo get` scripts in series also in a shared bgcval2 directory: 
    - ie `/gws/nopw/j04/esmeval/bgcval2/shared_mass_scripts/`
- It deletes jobID.sh scripts that are too old - ie we only want to download jobs that people are still interested in. This can be done with the `find` command.
- The `shared_mass_cron.sh` all has a self destruct after 23 hours. It will run every 24 hours, so we can't let it try to overlap with yesterdays run.

To do this, we'll need to: 
- [x] create new branch `automate_nightly_mass_download`
- [x] create `/gws/nopw/j04/esmeval/bgcval2/shared_mass_scripts/`
- [x] create `shared_mass_cron.sh`
  - [x] it runs everything in `shared_mass_scripts`
  - [x] delete old jobID.sh files in `shared_mass_scripts`.
  - [x] 23 hour killswitch. 
- [x] edit download_from_mass.py:
    - [x] add a line that copied the jobID.sh file to the shared central bgcval2 directory, but doesn't keep the original time stamp.
    - [x] Fix bug where path.py isn't used correctly.
- [x] Add a line to `input_yml` files and `analysis_compare` that allow this behaviour to be turned off if so desired.
- [x] edit `default-bgcval2-config.yml` `paths.py` to include new output path.
- [x] Add this to documentation and inform users. 
- [ ] Tell jasmin helpdesk about the nightly scipt.

I think this is everything, but if I think of anything else, I'll make sure to add it here, post hoc.

### PR DESCRIPTION
**Pull request by ledm**
A fairly complex set of changes that should allow us to automate nightly mass downloads. A lot more details in the issue #62.

Closes #62.

### CODE REVIEW

File: bgcval2/analysis_compare.py [hunk 1]
```diff
@@ -1672,16 +1672,16 @@ def load_comparison_yml(master_compare_yml_fn):
         Details dict.
     """
     with open(master_compare_yml_fn, 'r') as openfile:
-        dictionary = yaml.safe_load(openfile)
```
Comment (ledm) [inline bgcval2/analysis_compare.py:1675]: can't believe how long it took me to rename this variable.

**Comment by ledm** [comment]
Note that I have not added any tools to delete the local copy of these data, once they've been downloaded. Perhaps we could add a system for that in here too.

**Comment by valeriupredoi** [comment]
you wanna merge this branch, man? Or still working on it?

**Comment by ledm** [comment]
yeah, I think so! Should be mergeable...

image

File: bgcval2/download_from_mass.py [hunk 2]
```diff
@@ -486,6 +475,11 @@ def download_from_mass(jobID, doMoo=True):
     outfile.write(download_script_txt)
     outfile.close()
 
+    if auto_download:
+        shared_file_path = os.path.join(paths.shared_mass_scripts, os.path.basename(download_script_path))
+        print('writing file in shared path', shared_file_path)
+        shutil.copy(download_script_path, shared_file_path)
```
Comment (valeriupredoi) [inline bgcval2/download_from_mass.py:481]: make sure you also `rmtree` what's not needed

File: README.md [hunk 3]
```diff
@@ -206,6 +223,11 @@ both to download, but also to run the monthly analysis is not currently tested.
 This tool downloads the data, but also includes several functions which create symbolic links
 in the data's directory in order to accomodate incompatible changes in NEMO's output formatting.
 
+Please consult the command help for more details:
+```
+download_from_mass -h
```
Comment (valeriupredoi) [inline README.md:228]: can you add this to the commands we test in `.github/workflows/run-tests.yml` as well pls?

**Review by valeriupredoi — CHANGES_REQUESTED** [review]
noice! Couple tiny wiggles from me, man :beer:

**Review by valeriupredoi — CHANGES_REQUESTED** [review]

File: bgcval2/download_from_mass.py (hunk 2 above)
Comment (valeriupredoi) [reply bgcval2/download_from_mass.py:481]: nvmnd it's just a wee file

File: bgcval2/download_from_mass.py [hunk 4]
```diff
@@ -436,17 +421,21 @@ def download_from_mass(jobID, doMoo=True):
...
+            output = process.communicate()[0]
     else:
-        print("download_from_mass:\trunning the command:", bashCommand)
-        stdout.flush()
-        process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
-        output = process.communicate()[0]
+        output=''
 
     # moo get:
     if len(output.split('\n')) > 6000:
```
[the remark this answers is not in the document]
Comment (valeriupredoi) [reply bgcval2/download_from_mass.py:441]: fixed it now

File: README.md (hunk 3 above)
Comment (valeriupredoi) [reply README.md:228]: did it now

**Review by valeriupredoi — APPROVED** [review]
nice work @ledm :beer:

**Pull request merged**